jaxrs/src/main/java/org/jboss/as/jaxrs/ResteasyExtensionTransformerRegistration.java

```diff
@@ -0,0 +1,71 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.jaxrs;
+
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.transform.ExtensionTransformerRegistration;
+import org.jboss.as.controller.transform.SubsystemTransformerRegistration;
+import org.jboss.as.controller.transform.description.ChainedTransformationDescriptionBuilder;
+import org.jboss.as.controller.transform.description.DiscardAttributeChecker;
+import org.jboss.as.controller.transform.description.RejectAttributeChecker;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.as.controller.transform.description.TransformationDescriptionBuilder;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@MetaInfServices
+public class ResteasyExtensionTransformerRegistration implements ExtensionTransformerRegistration {
+
+    private static final ModelVersion VERSION_3_0_0 = ModelVersion.create(3, 0, 0);
+    private static final ModelVersion VERSION_2_0_0 = ModelVersion.create(2, 0, 0);
+    private static final ModelVersion VERSION_1_0_0 = ModelVersion.create(1, 0, 0);
+
+    @Override
+    public String getSubsystemName() {
+        return JaxrsExtension.SUBSYSTEM_NAME;
+    }
+
+    @Override
+    public void registerTransformers(final SubsystemTransformerRegistration subsystemRegistration) {
+        ChainedTransformationDescriptionBuilder builder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(subsystemRegistration.getCurrentSubsystemVersion());
+
+        registerV3Transformers(builder.createBuilder(JaxrsExtension.CURRENT_MODEL_VERSION, VERSION_3_0_0));
+        registerV1Transformers(builder.createBuilder(VERSION_2_0_0, VERSION_1_0_0));
+
+        builder.buildAndRegister(subsystemRegistration, new ModelVersion[] {VERSION_1_0_0, VERSION_2_0_0, VERSION_3_0_0, JaxrsExtension.CURRENT_MODEL_VERSION});
+    }
+
+    private static void registerV3Transformers(ResourceTransformationDescriptionBuilder subsystem) {
+        subsystem.getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD);
+    }
+
+    private static void registerV1Transformers(ResourceTransformationDescriptionBuilder subsystem) {
+        subsystem.getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, JaxrsAttribute.ATTRIBUTES)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, JaxrsAttribute.ATTRIBUTES);
```

Comment:
Using JaxrsAttribute.ATTRIBUTES here is not good. The transformation rules should not change without explicit thought if that list of attributes changes.

Reply:
I can definitely list them out. The only reason I did this was because the first iteration of the subsystem had no attributes.